README.rst

```diff
@@ -94,19 +101,267 @@ https://hub.docker.com/r/nethserver/makerpms.
+
+`travis-ci.com <https://travis-ci.com>`_ automatically builds RPMs and uploads
+them to ``packages.nethserver.org``, if configured with enough environment variables
+and upload secrets.
+
+Configuration
+^^^^^^^^^^^^^
+
+To automate the RPM build process using Travis CI
+
+* create a ``.travis.yml`` file inside the source code repository hosted on
+  GitHub.
+
+* the repository must have Travis CI builds enabled and upload secrets properly set up.
+  Contact the organization maintainer on community.nethserver.org for help.
+
+The list of enabled repositories is available at `NethServer page on
+travis-ci.com <https://travis-ci.com/NethServer/>`_.
+
+This is an example of ``.travis.yml`` contents: ::
+
+  ---
+  language: ruby
+  services:
+      - docker
+  branches:
+      only:
+          - master
+  env:
+    global:
+      - DEST_ID=core
+      - NSVER=7
+      - DOCKER_IMAGE=nethserver/makerpms:${NSVER}
+      - >
+          EVARS="
+          -e DEST_ID
+          -e TRAVIS_BRANCH
+          -e TRAVIS_BUILD_ID
+          -e TRAVIS_PULL_REQUEST_BRANCH
+          -e TRAVIS_PULL_REQUEST
+          -e TRAVIS_REPO_SLUG
+          -e TRAVIS_TAG
+          -e NSVER
+          -e ENDPOINTS_PACK
+          "
+  script: |
+        set -e
+        docker run -ti \
+          --name makerpms ${EVARS} \
+          --hostname "b${TRAVIS_BUILD_NUMBER}.nethserver.org" \
+          --volume $PWD:/srv/makerpms/src:ro ${DOCKER_IMAGE} \
+          makerpms-travis *.spec
+        docker commit makerpms nethserver/build
+        docker run -ti ${EVARS} \
+          -e SECRET \
+          -e SECRET_URL \
+          -e AUTOBUILD_SECRET \
+          -e AUTOBUILD_SECRET_URL \
+          nethserver/build uploadrpms-travis
+
+Usage
+^^^^^
+
+Travis CI builds are triggered automatically when:
+
+* one or more commits are pushed to the `master` branch of the NethServer repository, as
+  stated in the ``.travis.yml`` file above by the ``branches`` key
+
+* A *pull request* is opened from a NethServer repository fork or it is updated
+  by submitting new commits
+
+After a successful build, the RPM is uploaded to ``packages.nethserver.org``,
+according to the ``DEST_ID`` variable value. Supported values are ``core`` for
+NethServer core packages, and ``forge`` for NethForge packages.
+
+Pull requests are commented automatically by ``nethbot``
+[#NethBot]_ with the links to available RPMs.
+
+Also issues are commented by ``nethbot`` if the following rules are respected in git commit messages:
+
+1. The issue reference (e.g. ``NethServer/dev#1234``) is present in the merge
+   commit of pull requests
+
+2. The issue reference is added to standalone commits (should be rarely used)
+
+
+The build environment supports the following variables:
+
+- ``NSVER``
+- ``DOCKER_IMAGE``
+- ``DEST_ID``
+
+NSVER
+~~~~~
+
+``NSVER`` selects the target NethServer version for the build system. Currently
+the supported version values are ``7`` and ``6``.
+
+DOCKER_IMAGE
```

Comment:
The `ŧravis.yml` of this repo uses the `IMAGE_REPO` variable not `DOCKER_IMAGE`.
Which is the right one?

Reply:
They are two distinct things: 

- IMAGE_REPO is just a prefix used to build and push the container image to Docker hub.
- DOCKER_IMAGE is self-contained in the .travis.yml example for RPM builds.

Reply:
:man_shrugging: